ip_country/src/country_block_stream.rs

```diff
@@ -0,0 +1,411 @@
+use csv::{StringRecord, StringRecordIter};
+use std::cmp::Ordering;
+use std::net::{IpAddr, Ipv4Addr, Ipv6Addr};
+use std::str::FromStr;
+
+#[derive(Clone, PartialEq, Debug, Eq)]
+pub struct Country {
+    pub index: usize,
+    pub iso3166: String,
+    pub name: String,
+    pub free_world: bool,
+}
+
+impl Country {
+    pub fn new(index: usize, iso3166: &str, name: &str, free_world: bool) -> Self {
+        Self {
+            index,
+            iso3166: iso3166.to_string(),
+            name: name.to_string(),
+            free_world,
+        }
+    }
+}
+
+#[derive(Clone, PartialEq, Debug)]
+pub enum IpRange {
+    V4(Ipv4Addr, Ipv4Addr),
+    V6(Ipv6Addr, Ipv6Addr),
+}
+
+impl IpRange {
+    //TODO make 2 new functions "before", "after" to check if ip is in the range or before, or after
+    // ipV4 you can represent as u32 and ipV6 as u128
+    // can reuse guts of contains for before and after and then replace arms of match witch calls to those fns
+    pub fn contains(&self, ip_addr: IpAddr) -> bool {
+        match self {
+            IpRange::V4(begin, end) => match ip_addr {
+                IpAddr::V4(candidate) => Self::contains_inner(
+                    u32::from(*begin) as u128,
+                    u32::from(*end) as u128,
+                    u32::from(candidate) as u128,
+                ),
+                IpAddr::V6(_candidate) => false,
+            },
+            IpRange::V6(begin, end) => match ip_addr {
+                IpAddr::V4(_candidate) => false,
+                IpAddr::V6(candidate) => Self::contains_inner(
+                    u128::from(*begin),
+                    u128::from(*end),
+                    u128::from(candidate),
+                ),
+            },
+        }
+    }
+
+    pub fn in_range(&self, ip_addr: IpAddr) -> Ordering {
```

Comment:
This name doesn't fits. It evokes that this function should return a boolean, answering a question like "Is this IP **in** this **range**". Maybe `range_related_ordering`. These names are tough. Grrr.